### DISCUSSION TITLE
autoRefreshToken does not refresh access token

### DISCUSSION THREAD

**Discussion by stefan-girlich** [nhost/nhost#2970] [Q&A]
I am using `@nhost/nextjs": "^2.1.22"` with `"next": "14.2.4"`, initializing `NhostClient` with `autoRefreshToken: true`.
A user is signed in via `useSignInAnonymous` at the beginning of an onboarding flow and should keep their existing session indefinitely via `autoRefreshToken: true`.
`useAccessToken()` will provide a valid access token, but will continue to provide the same token even after it has expired;
is not replaced with a fresh one after its expiration date.

Is there something I am missing? Am I misunderstanding `autoRefreshToken`?

**Comment by onehassan** [comment]
Check out our [Next.js example](https://github.com/nhost/nhost/blob/main/examples/nextjs/pages/public-csr.tsx) which demonstrates the use of `useSignInAnonymous`. It should automatically refresh to display a new, valid accessToken when the current one expires. If you notice any differences between the example and your setup, let us know—that information can help us troubleshoot more effectively.
